librz/analysis/arch/arm/arm_il32.c

```diff
@@ -101,15 +137,108 @@ static RzILOpBitVector *read_reg(ut64 pc, arm_reg reg) {
 		RzILOpBitVector *var = VARG(reg_var_name(ARM_REG_D0 + idx / 2));
 		return UNSIGNED(32, idx % 2 ? SHIFTR0(var, UN(7, 32)) : var);
 	}
+	if (reg >= ARM_REG_Q0 && reg <= ARM_REG_Q15) {
+		ut32 low_dr_idx = (reg - ARM_REG_Q0) << 1;
+		ut32 high_dr_idx = low_dr_idx + 1;
+		RzILOpBitVector *low_var = VARG(reg_var_name(ARM_REG_D0 + low_dr_idx));
+		RzILOpBitVector *high_var = VARG(reg_var_name(ARM_REG_D0 + high_dr_idx));
+		return APPEND(high_var, low_var);
+	}
+
 	const char *var = reg_var_name(reg);
 	return var ? VARG(var) : NULL;
 }
 
+/**
+ * Return IL of bitvector store in register lane
+ * The length of such bitv is `data_size`
+ */
+static RzILOpBitVector *read_reg_lane(arm_reg reg, ut32 lane, ut32 data_size) {
+	if (is_core_reg(reg)) {
+		rz_warn_if_reached();
+		return NULL;
```

Comment:
Wouldn't it be better to use `rz_return_val_if_fail` here? Would be more concise.

Reply:
yep, would consider that

Reply:
It is generally nicer to have `rz_return_*` functions at the start of a function rather than `rz_warn_if_reached`. So try to replace it other such places as well. Thanks!